Rust/Rust.sublime-syntax

```diff
@@ -803,6 +794,31 @@ contexts:
     - match: '[-+%/*]'
       scope: keyword.operator.arithmetic.rust
 
+  impl-trait:
+    - match: '\bimpl\b'
+      scope: storage.type.impl.rust
+      push:
+        - match: '\buse\b'
+          scope: keyword.other.rust
+          push:
+            - match: '<'
+              scope: punctuation.definition.generic.begin.rust
+              set:
+                - match: '>'
+                  scope: punctuation.definition.generic.end.rust
+                  pop: true
+                - include: lifetime
+                - match: ','
+                  scope: punctuation.separator.rust
+                - match: \b(Self)\b
+                  scope: keyword.other.rust
+                - match: '{{identifier}}'
```

Comment:
`Self` is already defined as storage type in `no-path-type-names` context and identifiers should also receive valuable scope names. Types are currently unscoped as other identifiers are.

I'd therefore suggest to include ordinary `identifiers` context, which would also - even if maybe not required - handle fully qualified identifiers.

```suggestion
                - include: identifiers
```

Reply:
While it may not be 100% accurate either, I'd rather include `types` here since that is meant to be used in situations where types are expected. It matches lifetimes as well as stray identifiers and commonly known stdlib types. 

Commas still need to be matched separately, though.

Reply:
Actually, reading the spec again, pathed identifiers aren't supported and it is also unlikely for built-in names to be used since this is supposed to refer to generics and a generic with a built-in name would shadow it.

In that case, please instead simply include `all-generic-type-names`.